### PR TITLE
Improve source selection UX

### PR DESCRIPTION
**Pull request by hannahblair**
## Description

This PR

- Fixes the source selection being cut off in Webcam (and changes it to use the `select` and `option` elements so a little a11y win)
- Changes `gr.Image` to use the source selection component used in Audio and Video for alignment (extra benefit is seeing the selected source icon styling)
    - I've changed `active_source` logic slightly: 1) it can be any of the sources, not just `webcam` or `null`. 2) `active_tool` renamed to `active_source` for a tiny bit of consistency cc @pngwn 
- Adds a paste from clipboard view when `active_source` is clipboard
- Improve Video source selection styling (use webcam icon) and UX (couldn't switch to upload from webcam)

Test Demo

```
import gradio as gr

with gr.Blocks() as demo:
    with gr.Row():
        gr.Image(label="Input", sources=["upload", "webcam", "clipboard"])
        gr.Image(label="Input", sources=["webcam", "webcam"])
        gr.Video(sources=["webcam", "upload"])

demo.launch()

```

 | | Before | After |
|--------|--------|--------|
| `gr.Image` | Kapture 2023-12-13 at 12 29 17 | Kapture 2023-12-13 at 12 27 26
| `gr.Image` | Kapture 2023-12-13 at 12 12 13  | Kapture 2023-12-13 at 12 15 10 
| `gr.Video` |    Kapture 2023-12-13 at 13 25 57 | Kapture 2023-12-13 at 13 28 58
| `gr.Video` (can now be small in mobile) | Screenshot 2023-12-13 at 14 35 56 | Screenshot 2023-12-13 at 14 33 58

Closes: #6710
Closes: #6751 

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

**Comment by abidlabs** [comment]
Thanks @hannahblair! I'll mark it as a regression then

**Comment by abidlabs** [comment]
Great! Perhaps @dawoodkhan82 or @aliabid94 could review the PR

File: js/atoms/src/SelectSource.svelte [hunk 1]
```diff
@@ -1,22 +1,30 @@
...
-	export let sources: string[];
-	export let active_source: string;
+	type sources = "upload" | "microphone" | "webcam" | "clipboard" | null;
+
+	export let sources: Partial<sources>[];
+	export let active_source: Partial<sources>;
 	export let handle_clear: () => void = () => {};
+	export let handle_select: (source_type: Partial<sources>) => void = () => {};
+
+	let unique_sources = [...new Set(sources)];
```
Comment (aliabid94) [inline js/atoms/src/SelectSource.svelte:11]: not reactive, right?
```suggestion
	$: unique_sources = [...new Set(sources)];
```
(not completely sure if this deduplication logic is even necessary tbh? no particular strong feelnig about it tho)

File: js/atoms/src/SelectSource.svelte [hunk 2]
```diff
@@ -1,22 +1,30 @@
 <script lang="ts">
-	import { Microphone, Upload, Video } from "@gradio/icons";
+	import { Microphone, Upload, Webcam, ImagePaste } from "@gradio/icons";
 
-	export let sources: string[];
-	export let active_source: string;
+	type sources = "upload" | "microphone" | "webcam" | "clipboard" | null;
+
+	export let sources: Partial<sources>[];
```
Comment (aliabid94) [inline js/atoms/src/SelectSource.svelte:6]: why the use of Partial here and throughout this file? I don't believe it makes a difference in this context.
```suggestion
	export let sources: sources[];
```

Comment (aliabid94) [reply js/atoms/src/SelectSource.svelte:6]: also would name the type differently if we also have a variable called sources.

**Comment by aliabid94** [comment]
I found the new "paste from clipboard" UX a little confusing if I don't have an image in clipboard. previously, it would do nothing which makes sense. Now it shows text for "Paste your Image", which is just a file upload?

**Comment by pngwn** [comment]
Yeah, having paste as an active tool is a little misleading because it is a discreet action. It either works or it doesn't require any additional actions.

**Comment by dawoodkhan82** [comment]
Nice, the changes look clean! Should we have placeholder text in the dropdown if there are no sources?
Screenshot 2023-12-13 at 4 18 53 PM

Also unrelated to this PR, but it wasn't clear to me the dropdown arrow for sources, and the image capture/record button were two separate buttons. Since they're in the same button looking container.

Screenshot 2023-12-13 at 4 47 46 PM

**Comment by pngwn** [comment]
Thanks for those changes, will take a look!

**Review by pngwn — APPROVED** [review]
Looks good to me. Thanks @hannahblair!

**Pull request merged**